Payment/Program.cs

```diff
@@ -1,2 +1,183 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿namespace Payment;
+
+public class Program
+{
+	public static void Main()
+	{
+		Console.Write("Enter payment type (CreditCard/CryptoCurrency): ");
+		var input = Console.ReadLine();
+
+		if (!Enum.TryParse(input, true, out PaymentType paymentType))
+		{
+			paymentType = PaymentType.Unsupported;
+		}
+
+		Console.Write("Enter payment amount: ");
+		if (!decimal.TryParse(Console.ReadLine(), out var price))
+		{
+			Console.WriteLine("Invalid price. Please enter a valid number.");
+			return;
+		}
+
+		var payment = PaymentService.Create(paymentType, price);
+
+		var response = payment.ProcessPayment(price);
+
+		Console.WriteLine(response.Message);
+	}
+}
+
+public class PaymentService
+{
+	public static IPaymentService Create(PaymentType type, decimal price)
+	{
```

Comment:
Adding new payment methods requires modifications to the PaymentService.Create method. This violates the Open/Closed Principle (OCP) since the code is not open for extension without modification.